src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -1856,6 +2219,87 @@ private void cleanupEncodeSlotIfNecessary()
             }
         }
 
+        private final class AmqpAnnotations
+        {
+            private final AmqpMapFW.Builder<AmqpValueFW, AmqpValueFW, AmqpValueFW.Builder, AmqpValueFW.Builder> annotationRW =
+                new AmqpMapFW.Builder<>(new AmqpValueFW(), new AmqpValueFW(), new AmqpValueFW.Builder(),
+                    new AmqpValueFW.Builder());
+
+            private AmqpMapFW<AmqpValueFW, AmqpValueFW> annotations(
+                AmqpDataExFW dataEx,
+                int bufferOffset)
+            {
+                final ArrayFW<AmqpAnnotationFW> annotations = dataEx.annotations();
+                if (annotations.fieldCount() > 0)
+                {
+                    annotationRW.wrap(extraBuffer, bufferOffset, extraBuffer.capacity());
+                    annotations.forEach(this::annotation);
+                    return annotationRW.build();
+                }
+                return null;
+            }
+
+            private void annotation(
+                AmqpAnnotationFW item)
+            {
+                switch (item.key().kind())
+                {
+                case KIND_ID:
+                    AmqpULongFW id = amqpULongRW.wrap(valueBuffer, valueOffset.value, valueBuffer.capacity())
```

Comment:
To set each annotation item and application properties item, I'm currently using the same buffer (valueBuffer), and I'm concerned that if we don't use the global variable as an offset, it might be an issue when setting both annotation and application properties items. I think I had that issue before, so that is why I initially made the offset global, but now I don't have spec scripts that set both annotations and application properties at once so I don't have an error.

Reply:
It will still effectively be global since the merged `AmqpAnnotations` and `AmqpApplicationProperties` class instance will still be global, right?

Reply:
Oh I see!